flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -373,7 +384,9 @@ function stop_cluster {
   "$FLINK_DIR"/bin/stop-cluster.sh
 
   # stop zookeeper only if there are processes running
-  if ! [ "`jps | grep 'FlinkZooKeeperQuorumPeer' | wc -l`" = "0" ]; then
+  zookeeper_process_count=$(jps | grep -c 'FlinkZooKeeperQuorumPeer' || true)
```

Comment:
according to [this document](https://www.gnu.org/software/bash/manual/bashref.html#The-Set-Builtin) conditions may return non-zero exit codes without causing the script to exit.

Reply:
As far as I remember, this caused an error in my setup. I can verify it again.